pallets/usage/src/lib.rs

```diff
@@ -406,14 +407,27 @@ pub mod pallet {
 
                                 QueueOf::<T>::queue(dispatch)
                                     .unwrap_or_else(|e| unreachable!("Message queue corrupted! {:?}", e));
-                            } else if MailboxOf::<T>::insert(dispatch.into_parts().1).is_err() {
+                            } else {
+                                let message = match dispatch.exit_code() {
+                                    Some(0) | None => dispatch.into_parts().1,
+                                    _ => {
+                                        let message = dispatch.into_parts().1;
+                                        message
+                                            .clone()
+                                            .with_string_payload::<ExecutionErrorReason>()
+                                            .unwrap_or(message)
+                                    }
+                                };
```

Comment:
Maybe it is worth to extract the part to standalone function and place it somewhere in core/common?
```rust
fn to_message_with_string_payload_if_any(dispatch: Dispatch) -> Message {
    let exit_code = dispatch.exit_code();
    let (_, message, ..) = dispatch.into_parts();
    match exit_code {
        Some(0) | None => message,
        _ => message
                .clone()
                .with_string_payload::<ExecutionErrorReason>()
                .unwrap_or(message)
    }
}
```

Reply:
Discussed. 
I don't want to provide such api for core, especially due to the fact that that in #1083 we will use concrete type in error creation, instead of just bytes, so converting payload to string will be infallible